### ISSUE TITLE
[Bug]: Frequent `Unable to create '.git/index.lock': File exists.`

### ISSUE THREAD

**Issue by pdelre** [denolehov/obsidian-git#342]
### Describe the bug

I'm frequently experiencing the error below with the plugin enabled. It does not seem to affect the functionality as commits are still being generated, but it is noisy as it makes for a large Obsidian notification. It feels like a concurrency race condition given commits are happening at the same general time (milliseconds).

My feeling is this has increased recently around the Obsidian v1.0.0 release, but that could be biased with all the other non-obsidian-git changes.

### Relevant errors (if available)

```shell
Uncaught (in promise) Error: fatal: Unable to create '/path/to/vault/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.

    at Object.action (VM18586 plugin:obsidian-git:23891:25)
    at PluginStore.exec (VM18586 plugin:obsidian-git:23916:25)
    at eval (VM18586 plugin:obsidian-git:21503:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (VM18586 plugin:obsidian-git:21501:16)
    at GitExecutorChain.eval (VM18586 plugin:obsidian-git:21485:44)
    at Generator.next (<anonymous>)
    at fulfilled (VM18586 plugin:obsidian-git:20334:24)
action @ VM18586 plugin:obsidian-git:23891
exec @ VM18586 plugin:obsidian-git:23916
eval @ VM18586 plugin:obsidian-git:21503
handleTaskData @ VM18586 plugin:obsidian-git:21501
eval @ VM18586 plugin:obsidian-git:21485
fulfilled @ VM18586 plugin:obsidian-git:20334
```
```


### Steps to reproduce

Best I understand, it's occurs under standard usage where an auto backup will trigger it. For lack of anything else, the plugin options are below.

The repo itself is a local only repo (no remotes, no pushing) and the whole vault is synced with SyncThing.

```
{
  "commitMessage": "{{hostname}} manual backup: {{date}}",
  "autoCommitMessage": "{{hostname}} backup: {{date}}",
  "commitDateFormat": "YYYY-MM-DD HH:mm:ss",
  "autoSaveInterval": 2,
  "autoPushInterval": 0,
  "autoPullInterval": 0,
  "autoPullOnBoot": false,
  "disablePush": true,
  "pullBeforePush": true,
  "disablePopups": false,
  "listChangedFilesInMessageBody": false,
  "showStatusBar": true,
  "updateSubmodules": false,
  "syncMethod": "merge",
  "customMessageOnAutoBackup": false,
  "autoBackupAfterFileChange": true,
  "treeStructure": false,
  "refreshSourceControl": true,
  "basePath": "",
  "differentIntervalCommitAndPush": false,
  "changedFilesInStatusBar": false,
  "username": "",
  "showedMobileNotice": true,
  "refreshSourceControlTimer": 7000,
  "showBranchStatusBar": true
}
```

### Expected Behavior

_No response_

### Addition context

_No response_

### Operating system

macOS

### Plugin version

2.7.0

**Comment by Vinzent03** [comment]
So it happens on auto backup without you triggering another git command?

**Comment by pdelre** [comment]
Correct. Best I can figure I'm not using any additional tools (IDEs, etc) that would cause Git to set a lock. It's possible its the background `git gc`, but given the frequency and pairing with a successful notification/toast, it feels like multiple Promises (or whatever concurrency objects) are created, conflict, and some succeed and others fail.

**Comment by Vinzent03** [comment]
Sorry, but I can't reproduce and don't know where exactly the bug is

**Comment by pdelre** [comment]
Ya, I figured this would be a tricky one. I haven't noticed it happen today 🤷.

I'm also happy to troubleshoot in DevTools if you have any resources/pointers on the best way to do so.

**Comment by pdelre** [comment]
I haven't seen an errors in notifications or the Console. I did do a forced reload and a close-reopen so it might have been a bad state that was fallen into.

I'll close the issue and try to get more information if it comes up again, including verifying the above resolution.

**Comment by mahoromax** [comment]
I'm experiencing the same behaviour - kind of frequent errors, and I only noticed it after I updated to Obsidian v1.0
Whenever I check to delete the file, it's not there.

I am not running any other sync software, or have any other git utility running, so I also think it's obsidian-git tripping over its own feet.

I could find the error in the terminal.
```
Uncaught (in promise) Error: fatal: Unable to create 'C:/myvaultpath/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.

    at Object.action (plugin:obsidian-git:23900:25)
    at PluginStore.exec (plugin:obsidian-git:23925:25)
    at eval (plugin:obsidian-git:21512:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (plugin:obsidian-git:21510:16)
    at GitExecutorChain.eval (plugin:obsidian-git:21494:44)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-git:20343:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

might be worth reopening...

**Comment by avh4** [comment]
I've noticed this happening a lot in the past week also.  Obsidian v1.0.3, MacOS, obsidian-git v2.10.0.
